### PR TITLE
moved secrets to be part of cloudzero config, updated documentation to mathc

### PR DESCRIPTION
**Pull request by roberthocking**
By submitting a PR to this repository, you agree to the terms within the [CloudZero Code of Conduct](https://github.com/cloudzero/template-cloudzero-open-source/blob/master/CODE-OF-CONDUCT.md). Please see the [contributing guidelines](https://github.com/cloudzero/template-cloudzero-open-source/blob/master/CONTRIBUTING.md) for how to create and submit a high-quality PR for this repo.

### Description

> Describe the purpose of this PR along with any background information and the impacts of the proposed change. For the benefit of the community, please do not assume prior context.
>
> Provide details that support your chosen implementation, including: breaking changes, alternatives considered, changes to the API, etc.
>
> If the UI is being changed, please provide screenshots.

### References

> Include any links supporting this change such as a:
>
> - GitHub Issue/PR number addressed or fixed
> - StackOverflow post
> - Support forum thread
> - Related pull requests/issues from other repos
>
> If there are no references, simply delete this section.

### Testing

> Describe how this can be tested by reviewers. Be specific about anything not tested and reasons why. If this library has unit and/or integration testing, tests should be added for new functionality and existing tests should complete without errors.
>
> Please include any manual steps for testing end-to-end or functionality not covered by unit/integration tests.
>
> Also include details of the environment this PR was developed in (language/platform/browser version).

- [ ] This change adds test coverage for new/changed/fixed functionality

### Checklist

- [ ] I have added documentation for new/changed functionality in this PR
- [ ] All active GitHub checks for tests, formatting, and security are passing
- [ ] The correct base branch is being used, if not `main`

### CODE REVIEW

File: charts/cloudzero-prometheus-agent/README.md [hunk 1]
```diff
@@ -29,10 +29,10 @@ The chart can be installed directly with Helm or any other common Kubernetes dep
 
 If installing with Helm directly, the following command will install the chart:
 ```console
-helm install [RELEASE_NAME] cloudzero/cloudzero-prometheus-agent \
-    --set api_key=CLOUDZERO_API_KEY \
-    --set cloudzero.cluster_name=CLUSTER_NAME \
-    --set cloudzero.cloud_account_id=CLOUD_ACCOUNT_ID \
+helm install <RELEASE_NAME> cloudzero/cloudzero-prometheus-agent \
```
Comment (dmepham) [inline charts/cloudzero-prometheus-agent/README.md:32]: the chart name is just `cloudzero-agent` now. sorry, I think I forgot to update that when I changed this

File: charts/cloudzero-prometheus-agent/README.md [hunk 2]
```diff
@@ -78,12 +78,12 @@ See the `kube-state-metrics` [documentation](https://github.com/kubernetes/kube-
...
-| cloudzero.cloud_account_id | string | `nil` | Account ID of the account the cluster is running in. |
-| cloudzero.cluster_name | string | `nil` | Name of the clusters. |
+| cloudzero.cloudAccountId | string | `nil` | Account ID of the account the cluster is running in. |
+| cloudzero.clusterName | string | `nil` | Name of the clusters. |
 | cloudzero.host | string | `"api.cloudzero.com"` | CloudZero host to send metrics to. |
-| credentials.createSecret | bool | `true` | If true, a secret containing the CloudZero API key will be created using the `credentials.api_key` value. |
-| credentials.api_key | string | `nil` | The CloudZero API key to use to export metrics. Only used if `credentials.createSecret` is true |
-| credentials.secretName | string | `""` | The name of the secret that contains the CloudZero API key. Required if createSecret is false. |
+| cloudzero.createSecret | bool | `true` | If true, a secret containing the CloudZero API key will be created using the `credentials.apiKey` value. |
+| cloudzero.apiKey | string | `nil` | The CloudZero API key to use to export metrics. Only used if `credentials.createSecret` is true |
```
Comment (dmepham) [inline charts/cloudzero-prometheus-agent/README.md:85]: I think this is still using the `credentials.` bit in the description here

File: charts/cloudzero-prometheus-agent/templates/_helpers.tpl [hunk 3]
```diff
@@ -14,13 +14,13 @@ Create chart name and version as used by the chart label.
...
 {{- define "cloudzero-agent.remoteWriteUrl" -}}
-https://{{ .Values.global.cloudzeroHost }}/v1/container-metrics?cluster_name={{.Values.global.cluster_name}}&cloud_account_id={{.Values.global.cloud_account_id}}
+https://{{ .Values.global.cloudzeroHost }}/v1/container-metrics?clusterName={{.Values.global.clusterName}}&cloudAccountId={{.Values.global.cloudAccountId}}
 {{- end -}}
 
 
 {{/* Define the secret name which holds the CloudZero API key */}}
 {{ define "cloudzero-agent.secretName" -}}
-{{ .Values.credentials.secretName | default (printf "%s-api-key" .Release.Name) }}
+{{ .Values.secretName | default (printf "%s-api-key" .Release.Name) }}
```
Comment (dmepham) [inline charts/cloudzero-prometheus-agent/templates/_helpers.tpl:23]: isn't this in the `cloudzero` block now?

File: charts/cloudzero-prometheus-agent/templates/_helpers.tpl [hunk 4]
```diff
@@ -14,13 +14,13 @@ Create chart name and version as used by the chart label.
 
 {{/* Define the url to which metrics should be sent */}}
 {{- define "cloudzero-agent.remoteWriteUrl" -}}
-https://{{ .Values.global.cloudzeroHost }}/v1/container-metrics?cluster_name={{.Values.global.cluster_name}}&cloud_account_id={{.Values.global.cloud_account_id}}
+https://{{ .Values.global.cloudzeroHost }}/v1/container-metrics?clusterName={{.Values.global.clusterName}}&cloudAccountId={{.Values.global.cloudAccountId}}
```
Comment (dmepham) [inline charts/cloudzero-prometheus-agent/templates/_helpers.tpl:17]: sorry, I think I added this named template but didn't use it. we should either get rid of it or use it in the configmap

File: charts/cloudzero-prometheus-agent/templates/cm.yaml [hunk 5]
```diff
@@ -136,7 +136,7 @@ data:
           follow_redirects: true
           enable_http2: true
     remote_write:
-      - url: 'https://{{ .Values.cloudzero.host }}/v1/container-metrics?cluster_name={{.Values.cloudzero.cluster_name}}&cloud_account_id={{.Values.cloudzero.cloud_account_id}}'
+      - url: 'https://{{ .Values.host }}/v1/container-metrics?clusterName={{.Values.clusterName}}&cloudAccountId={{.Values.cloudAccountId}}'
```
Comment (dmepham) [inline charts/cloudzero-prometheus-agent/templates/cm.yaml:139]: are `clusterName` and `cloudAccountId` for the params correct? I think we'd want to keep those. we didn't change that in the query params we set in our API

File: charts/cloudzero-prometheus-agent/templates/secret.yaml [hunk 6]
```diff
@@ -1,6 +1,6 @@
-{{- if .Values.credentials.createSecret -}}
-{{- if not .Values.credentials.api_key }}
-{{- fail "api_key must be provided if API key is not provided by an existing secret" }}
+{{- if .Values.createSecret -}}
+{{- if not .Values.apiKey }}
```
Comment (dmepham) [inline charts/cloudzero-prometheus-agent/templates/secret.yaml:2]: I think this goes in the `cloudzero` block

File: charts/cloudzero-prometheus-agent/templates/secret.yaml [hunk 7]
```diff
@@ -11,5 +11,5 @@ metadata:
   name: {{ include "cloudzero-agent.secretName" .}}
   namespace: {{ include "cloudzero-agent.namespace" . }}
 data:
-  value: {{ .Values.credentials.api_key | b64enc | quote }}
+  value: {{ .Values.apiKey | b64enc | quote }}
```
Comment (dmepham) [inline charts/cloudzero-prometheus-agent/templates/secret.yaml:14]: same here - are we putting this in the `cloudzero` block? maybe I just don't know what we decided on

**Review by dmepham — CHANGES_REQUESTED** [review]
I think some of my comments don't apply, as I'm not sure what the final decision was on where `apiKey` should live

File: charts/cloudzero-prometheus-agent/README.md [hunk 8]
```diff
@@ -81,9 +81,9 @@ See the `kube-state-metrics` [documentation](https://github.com/kubernetes/kube-
 | cloudzero.cloud_account_id | string | `nil` | Account ID of the account the cluster is running in. |
 | cloudzero.cluster_name | string | `nil` | Name of the clusters. |
 | cloudzero.host | string | `"api.cloudzero.com"` | CloudZero host to send metrics to. |
-| credentials.createSecret | bool | `true` | If true, a secret containing the CloudZero API key will be created using the `credentials.api_key` value. |
-| credentials.api_key | string | `nil` | The CloudZero API key to use to export metrics. Only used if `credentials.createSecret` is true |
-| credentials.secretName | string | `""` | The name of the secret that contains the CloudZero API key. Required if createSecret is false. |
+| cloudzero.createSecret | bool | `true` | If true, a secret containing the CloudZero API key will be created using the `credentials.api_key` value. |
+| cloudzero.api_key | string | `nil` | The CloudZero API key to use to export metrics. Only used if `credentials.createSecret` is true |
```
[the remark this answers is not in the document]
Comment (roberthocking) [reply charts/cloudzero-prometheus-agent/README.md]: Fixed

File: charts/cloudzero-prometheus-agent/values.yaml [hunk 9]
```diff
@@ -5,15 +5,14 @@ cloudzero:
   cloud_account_id: null
   # -- Name of the clusters.
   cluster_name: null
-
-credentials:
```
[the remark this answers is not in the document]
Comment (roberthocking) [reply charts/cloudzero-prometheus-agent/values.yaml:9]: Moved them to top level settings. We can revisit this based on additional feedback, but it's good to go now.

File: charts/cloudzero-prometheus-agent/templates/secret.yaml (hunk 7 above)
Comment (roberthocking) [reply charts/cloudzero-prometheus-agent/templates/secret.yaml:14]: We removed the cloudzero block

File: charts/cloudzero-prometheus-agent/templates/secret.yaml (hunk 6 above)
Comment (roberthocking) [reply charts/cloudzero-prometheus-agent/templates/secret.yaml:2]: We removed the cloudzero block

File: charts/cloudzero-prometheus-agent/templates/cm.yaml (hunk 5 above)
Comment (roberthocking) [reply charts/cloudzero-prometheus-agent/templates/cm.yaml:139]: Ooops.

File: charts/cloudzero-prometheus-agent/values.yaml (hunk 9 above)
[the remark this answers is not in the document]
Comment (dmepham) [reply charts/cloudzero-prometheus-agent/values.yaml:9]: great, yeah I like that format

File: charts/cloudzero-prometheus-agent/README.md (hunk 2 above)
Comment (roberthocking) [reply charts/cloudzero-prometheus-agent/README.md]: removed all references to the credentials in the description

File: charts/cloudzero-prometheus-agent/templates/_helpers.tpl (hunk 4 above)
Comment (roberthocking) [reply charts/cloudzero-prometheus-agent/templates/_helpers.tpl:17]: let's remove it.

**Review by dmepham — APPROVED** [review]
lgtm, just some minor things. we can also merge this and fix in a follow up PR if you'd prefer that

File: charts/cloudzero-agent/README.md [hunk 10]
```diff
@@ -29,10 +29,10 @@ The chart can be installed directly with Helm or any other common Kubernetes dep
 
 If installing with Helm directly, the following command will install the chart:
 ```console
-helm install [RELEASE_NAME] cloudzero/cloudzero-prometheus-agent \
-    --set api_key=CLOUDZERO_API_KEY \
-    --set cloudzero.cluster_name=CLUSTER_NAME \
-    --set cloudzero.cloud_account_id=CLOUD_ACCOUNT_ID \
+helm install <RELEASE_NAME> cloudzero/cloudzero-agent \
+    --set apiKey=<CLOUDZERO_apiKey> \
```
[the remark this answers is not in the document]
Comment (roberthocking) [reply charts/cloudzero-agent/README.md:33]: fixed

File: charts/cloudzero-agent/README.md [hunk 11]
```diff
@@ -45,7 +45,7 @@ If using a Secret external to this chart for the API key, ensure the Secret is c
 
 ```yaml
 data:
-  value: <API_KEY>
+  value: <apiKey>
```
[the remark this answers is not in the document]
Comment (roberthocking) [reply charts/cloudzero-agent/README.md:48]: fixed

File: charts/cloudzero-agent/values.schema.json [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+              "apiKey"
+          ]
+      }
+    }
+},
+"required": [
+    "host",
+    "clusterName",
+    "cloudAccountId",
+    "apiKey"
```
[the remark this answers is not in the document]
Comment (roberthocking) [reply charts/cloudzero-agent/values.schema.json:36]: Makes sense. I'll remove it and we can gather feedback and see what works best here.

**Pull request merged**